From b65fa5f8c0a7be096e8b9860e28310f65b0dbd59 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Tue, 12 May 2015 22:10:08 +0100
Subject: [PATCH] Fix enormous path_exists issue (broken symlinks..)

---
 gl-driver-switch.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gl-driver-switch.c b/gl-driver-switch.c
index 34d2884..c15ae0c 100644
--- a/gl-driver-switch.c
+++ b/gl-driver-switch.c
@@ -37,7 +37,7 @@ static inline const char *usage(void)
 static inline bool path_exists(const char *p)
 {
         __attribute__ ((unused)) struct stat st = {0};
-        return ((stat(p, &st) == 0));
+        return ((lstat(p, &st) == 0));
 }
 
 static inline void _frees_(void *p)
-- 
2.3.2

